### DISCUSSION TITLE
retrive tab info such as tabID, tab label name, directly from component

### DISCUSSION THREAD

**Discussion by scalpovich** [soybeanjs/soybean-admin#383] [General]
[Info]: Hi, How to retrive tab info such as tabID, tab label name, directly from component generate by route, also how to manage display same multi tab with different state?

**Comment by honghuangdc** [comment]
1. in tab store shared, function `findTabByRouteName` can do it, you can expose function in tab store which only need param `routeName`.
2. configurate `multi: true` in route's meta, then with different query, the route will display in multi tab.

**Reply by scalpovich** [reply]
thanks for your reply, 
assuming i want to retrieve by default in all tab render component the tab id to generate uniq elements with uniq id such as div id , to ensure that for multi tab, with same content, when change the state its will not affect the other multi tab with same key  . Because right now even i use different query, when i update data in a specific tab with query param, it automatically affect others tab woth same key. Seems like cache logic, just cached key or tab name not the specific tab id. Can you provide directives to overrides?
